### PR TITLE
Add session by ID API Endpoint

### PR DESCRIPTION
**Pull request by PineappleIOnic**
## What does this PR do?
This PR adds a Extra API Endpoint to get a session based on it's ID or get the current one by using 'current' for Clients

It can be accessed using `GET /v1/account/sessions/:sessionId`

## Related PRs and Issues

This should help resolve this issue: https://github.com/appwrite/appwrite/issues/1009

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
Yes, I have

### CODE REVIEW

File: app/controllers/api/account.php [hunk 1]
```diff
@@ -968,6 +969,46 @@
...
+    ->action(function ($sessionId, $response, $user, $locale, $projectDB) {
+        /** @var Appwrite\Utopia\Response $response */
+        /** @var Appwrite\Database\Document $user */
+        /** @var Utopia\Locale\Locale $locale */
+
+        $sessionId = ($sessionId === 'current')
+        ? Auth::sessionVerify($user->getAttribute('sessions'), Auth::$secret)
+        : $sessionId;
+
+        $session = $projectDB->getCollectionFirst([ // Get user by email address
```
Comment (TorstenDittmann) [inline app/controllers/api/account.php:997]: We need to add the country name manually, since its not saved in the database.

You can find an example here: https://github.com/appwrite/appwrite/blob/master/app/controllers/api/account.php#L881-L883

File: app/controllers/api/account.php [hunk 2]
```diff
@@ -892,6 +892,7 @@
         ]), Response::MODEL_SESSION_LIST);
     });
 
+
```
Comment (TorstenDittmann) [inline app/controllers/api/account.php:895]: Unnecessary whitespace :+1:

File: app/controllers/api/account.php [hunk 3]
```diff
@@ -968,6 +969,46 @@
...
     });
 
+App::get('/v1/account/sessions/:sessionId')
+    ->desc('Get Session By ID')
+    ->groups(['api', 'account'])
+    ->label('scope', 'account')
+    ->label('sdk.auth', [APP_AUTH_TYPE_SESSION, APP_AUTH_TYPE_JWT])
+    ->label('sdk.namespace', 'account')
+    ->label('sdk.method', 'getSessions')
+    ->label('sdk.description', '/docs/references/account/get-sessions.md')
```
Comment (TorstenDittmann) [inline app/controllers/api/account.php:979]: This needs to be adjusted and add a new file to document it.

File: app/controllers/api/account.php [hunk 4]
```diff
@@ -968,6 +969,46 @@
         $response->dynamic(new Document(['logs' => $output]), Response::MODEL_LOG_LIST);
     });
 
+App::get('/v1/account/sessions/:sessionId')
+    ->desc('Get Session By ID')
+    ->groups(['api', 'account'])
+    ->label('scope', 'account')
+    ->label('sdk.auth', [APP_AUTH_TYPE_SESSION, APP_AUTH_TYPE_JWT])
+    ->label('sdk.namespace', 'account')
+    ->label('sdk.method', 'getSessions')
```
Comment (TorstenDittmann) [inline app/controllers/api/account.php:978]: This is already been used. Make it `getSession` 👍🏻

File: app/controllers/api/account.php [hunk 5]
```diff
@@ -968,6 +969,46 @@
...
+    ->label('sdk.description', '/docs/references/account/get-sessions.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_SESSION_LIST)
```
Comment (TorstenDittmann) [inline app/controllers/api/account.php:982]: Now we don't return a list of sessions anymore, but a single session. So need to be adjusted as well 👍🏻

File: app/controllers/api/account.php [hunk 6]
```diff
@@ -968,6 +969,46 @@
...
+    ->param('sessionId', null, new UID(), 'Session unique ID. Use the string \'current\' to get the current device session.')
+    ->inject('response')
+    ->inject('user')
+    ->inject('locale')
+    ->inject('projectDB')
+    ->action(function ($sessionId, $response, $user, $locale, $projectDB) {
```
Comment (TorstenDittmann) [inline app/controllers/api/account.php:988]: Do we need `$locale` here?

**Review by TorstenDittmann — CHANGES_REQUESTED** [review]

File: app/controllers/api/account.php (hunk 1 above)
Comment (christyjacob4) [inline app/controllers/api/account.php:997]:
```suggestion
        $session = $projectDB->getCollectionFirst([ // Get user by sessionId
```

File: app/controllers/api/account.php (hunk 6 above)
Comment (PineappleIOnic) [reply app/controllers/api/account.php:987]: We do now because we implement this https://github.com/appwrite/appwrite/pull/1294#discussion_r652555499

Comment (TorstenDittmann) [reply app/controllers/api/account.php:987]: Yes 👍🏻

File: app/controllers/api/account.php (hunk 1 above)
Comment (eldadfux) [reply app/controllers/api/account.php]: @PineappleIOnic the reason we don't have the country in the DB is that we support changing the country names to multiple languages.

Comment (eldadfux) [reply app/controllers/api/account.php]: For fetching a single document we should use `getDocument` as it has much better performance and uses the Redis cache. `getCollectionFirst` doesn't leverage caching.

File: app/controllers/api/account.php [hunk 7]
```diff
@@ -993,15 +993,9 @@
...
-            'limit' => 1,
-            'filters' => [
-                '$collection='.Database::SYSTEM_COLLECTION_SESSIONS,
-                '$id='.$sessionId,
-            ],
-        ]);
+        $session = $projectDB->getDocument($sessionId); // get user by session ID
 
-        if ($session == false) {
+        if (empty($session->getId()) || Database::SYSTEM_COLLECTION_SESSIONS != $session->getCollection()) {
```
Comment (eldadfux) [inline app/controllers/api/account.php:998]:
```suggestion
        if ($session->isEmpty() || Database::SYSTEM_COLLECTION_SESSIONS != $session->getCollection()) {
```

File: app/controllers/api/account.php [hunk 8]
```diff
@@ -968,6 +968,44 @@
...
+    ->label('sdk.response.model', Response::MODEL_SESSION)
+    ->param('sessionId', null, new UID(), 'Session unique ID. Use the string \'current\' to get the current device session.')
+    ->inject('response')
+    ->inject('user')
+    ->inject('locale')
+    ->inject('projectDB')
+    ->action(function ($sessionId, $response, $user, $locale, $projectDB) {
+        /** @var Appwrite\Utopia\Response $response */
+        /** @var Appwrite\Database\Document $user */
+        /** @var Utopia\Locale\Locale $locale */
```
Comment (eldadfux) [inline app/controllers/api/account.php:990]: Missing a comment for `$projectDB` to have IDE autocompletion

**Review by TorstenDittmann — APPROVED** [review]

**Review by eldadfux — APPROVED** [review]

File: app/controllers/api/account.php [hunk 9]
```diff
@@ -968,6 +968,45 @@
...
+        ? Auth::sessionVerify($user->getAttribute('sessions'), Auth::$secret)
+        : $sessionId;
+
+        $session = $projectDB->getDocument($sessionId); // get user by session ID
+
+        if ($session->isEmpty() || Database::SYSTEM_COLLECTION_SESSIONS != $session->getCollection()) {
+            throw new Exception('Session not found', 404);
+        };
+
+        $session->setAttribute('countryName', (isset($countries[strtoupper($session->getAttribute('countryCode'))]))
```
Comment (christyjacob4) [inline app/controllers/api/account.php:1003]: For better readability, i'd bring the ternary operator out of the `setAttribute` and store it in a variable
```php
$countryName = ...;
$session->setAttribute('countryName', $countryName);
```

**Comment by christyjacob4** [comment]
You might wanna add this to the change log and also change the base branch to `0.9.x`

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Please address the comments

**Comment by christyjacob4** [comment]
@PineappleIOnic Dont forget to add this to the changelog.
Take a look at this example https://github.com/appwrite/appwrite/pull/1142/files#diff-d975bf659606195d2165918f93e1cf680ef68ea3c9cab994f033705fea8238b2L25

**Review by eldadfux — APPROVED** [review]

**Review by christyjacob4 — APPROVED** [review]
Looks Good :+1:
